homeassistant/util/package.py

```diff
@@ -25,7 +25,7 @@ def is_virtual_env() -> bool:
 
 def is_docker_env() -> bool:
     """Return True if we run in a docker env."""
-    return Path("/.dockerenv").exists()
+    return Path("/.dockerenv").exists() and os.geteuid() == 0
```

Comment:
Regardless if the goal of this PR is accepted or not, this implementation is wrong. You are breaking all checks if it is a Docker environment, regardless of how the result is being used.

Reply:
Thanks for pointing that out and I probably should have renamed the function in my PR.

The function is only used in two places: 

1. In `requirements.py` (result stored in `is_docker`):
https://github.com/home-assistant/home-assistant/blob/f540d74b654700f349303cca7df0e13ba66a824a/homeassistant/requirements.py#L93-L94 and

2. in `config.py`:
https://github.com/home-assistant/home-assistant/blob/f540d74b654700f349303cca7df0e13ba66a824a/homeassistant/config.py#L400-L405

That is, as of now, this function is only ever used to introduce the behavior that has caused problems to users running the docker container as non-root.

Maybe the (new) function should be renamed to reflect that (`is_docker_env_with_root`? I'm terrible at naming) or maybe there should be a configuration setting that controls pip's install target and the logic should be removed completely?